server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1473,6 +1473,31 @@ public GatedCloseable<IndexCommit> acquireLastIndexCommit(boolean flushFirst) th
         }
     }
 
+    public GatedCloseable<IndexCommit> acquireLastIndexCommitAndRefresh(boolean flushFirst) throws EngineException {
+        GatedCloseable<IndexCommit> indexCommit = acquireLastIndexCommit(flushFirst);
+        getEngine().refresh("Snapshot for Remote Store based Shard");
```

Comment:
we will not need stats call, as we will look for the snapshot commit in remote store, it will present in remote store only if upload was successful, if we don't find that file, we will fail the snapshot.